### ISSUE TITLE
App stuck when trying to get the getAdServicesAttributionToken

### ISSUE THREAD

**Issue by kbihani** [joel-bitar/react-native-apple-ads-attribution#3]
Hi,

Thanks for creating this library
But when I am trying to use it with my app, the app gets stuck as soon as its hitting the function `getAdServicesAttributionToken`:
`NSString *attributionToken = [AAAttributionClass attributionTokenWithError:error];`
in AppleAdsAttribution.m

Any idea what might be going wrong? I don't see any kind of errors anywhere

**Comment by joel-bitar** [comment]
Hi,

The call to AAAttributionC attributionTokenWithError may block for a while if there is a delay from Apple (or if you are on bad internet connection). But I'm not sure how it can block in your AppDelegate.m, do you call the library directly from your AppDelegate? The library is meant to be used from javascript, with the exported RN-methods. See the readme. I haven't tested calling stuff directly from Objective C, so I'm not sure how that would work.

**Comment by kbihani** [comment]
Hi Joel,

No, I didn't make the call directly from Appdelegate.m
I tried calling from JS side. But when I saw the app was getting stuck consistently, I tried putting some debug logs in appropriate methods of `AppleAdsAttribution.m`
Thats where I saw the app getting stuck when trying to get the token. Internet works as everything else in the app works. 
Could it be that this call takes a lot of time for first load? Its blocking main thread as far as I could see.
App was consistently blocked for more than 10 minutes before I had to quit and restart.

**Comment by joel-bitar** [comment]
What version are you running, I noticed I forgot to publish 1.0.0 to npm. I published 1.0.1 now (no changes from 1.0.0, just fucked up publishing 😄).
Also I haven't tested the library in a Simulator since our app doesn't run in a simulator. Is it on a simulator you're getting problems? Also when in the app are you calling it (guess it shouldn't really matter, but we don't call it instantly on startup, but later on from a component). FYI we are on Xcode 13.1 and RN 0.66.1 and have tested it on a couple of different devices mostly devices running iOS 15. We are only using the
`const attributionData = await AppleAdsAttribution.getAttributionData()` though, but we are getting attribution data from the adservices api, so the token must be generated.

There is a problem with us sometimes getting 404 from apple though when using the token to get the attribution data. I will make a fix that retries requests that get 404 (I saw this is recommended by apple) at some point after the holidays.

**Comment by kbihani** [comment]
Hahaha yeah, I saw it was not published so just took your code, modified bits and created a native module.
Yeah I was trying on a Simulator. I'll test it on a device tonight, but yeah, it's getting stuck in a simulator.
Yes, I make the call from a component inside the app and not as soon as app starts.
I am trying with these configs:
**RN:** 0.63.4
**XCode:** 12.5
**Simulator:** iOS 14.5

I think calls might be getting rejected. Getting this in XCode logs after waiting on the stuck app for long time:
```
2021-12-23 15:57:18.844895+0530 xxxx[58035:477708] [connection] nw_socket_handle_socket_event [C16.1:3] Socket SO_ERROR [54: Connection reset by peer]
2021-12-23 15:57:18.845192+0530 xxxx[58035:477708] [connection] nw_read_request_report [C16] Receive failed with error "Connection reset by peer"
```

**Comment by joel-bitar** [comment]
Ah found someone with similar problems using Apples AAAttribution attributionTokenWithError API:
https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/issues/1109 and https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/issues/1112

They are also mentioning Simulator running iOS 14.5 and using Xcode 12.5. Hopefully it will work for you on real device. They solved it by adding a timeout to the request and also not check for token in Simulator. Maybe something like that is needed (https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/commit/84e9c553bee06c18db3c2dae938a3ef76c347357#diff-df1115018b3a68f52d5d012f60d0e3398527efc6baf6d097897ac4b60dbf82d2)

**Comment by kbihani** [comment]
Ah. Thank you sir! I'll try it on a device.
Could we integrate the isSimulator check to the library?

**Comment by joel-bitar** [comment]
Yeah, I will add a check for it. But I won't have time to do it until after the holidays.

**Comment by joel-bitar** [comment]
Should be fixed in 1.0.2.
